### PR TITLE
change: update metrics for scan and upload status

### PR DESCRIPTION
**Pull request by abhstabs**
Hi @ojaswa1942 / @khemrajrathore , 

added metrics for upload and scan status. Please have a look. 

Thank you. 

cc: @pandurangpatil

### CODE REVIEW

File: src/main/scala/ai/privado/entrypoint/ScanProcessor.scala [hunk 1]
```diff
@@ -328,6 +331,7 @@ object ScanProcessor extends CommandProcessor {
       case Failure(exception) => {
         logger.error("Error while parsing the source code!")
         logger.debug("Error : ", exception)
+        MetricHandler.setScanStatus(false)
         Left("Error while parsing the source code: " + exception.toString)
```
Comment (ojaswa1942) [inline src/main/scala/ai/privado/entrypoint/ScanProcessor.scala:335]: Do we capture this error message outside of this function?
If not, we should capture the error too

Comment (abhstabs) [reply src/main/scala/ai/privado/entrypoint/ScanProcessor.scala:335]: We capture the error outside of this function so it's not included here.

Comment (ojaswa1942) [reply src/main/scala/ai/privado/entrypoint/ScanProcessor.scala:335]: @abhstabs Should not both metrics, for one thing, be handled together.. if that makes the code cleaner?

Comment (abhstabs) [reply src/main/scala/ai/privado/entrypoint/ScanProcessor.scala:335]: The errors are logged one by one as they occur on a lower level. This is an umbrella exception to handle the user side messages.

Comment (ojaswa1942) [reply src/main/scala/ai/privado/entrypoint/ScanProcessor.scala:335]: @abhstabs Makes sense. 
Just want to ensure that even "unhandled exceptions" that are raised from lower-level functions are caught, which might not be handled anywhere else but only in the top handler. Looks good if you can confirm that.

Comment (abhstabs) [reply src/main/scala/ai/privado/entrypoint/ScanProcessor.scala:335]: Can confirm. Scan errors and upload errors are reported in the metrics at the lower level.

**Review by ojaswa1942 — APPROVED** [review]
Since all unhandled exceptions are also reported, as confirmed by @abhstabs, looks good.

**Comment by pandurangpatil** [comment]
@khemrajrathore have look at this one and if looks good to you go ahead and merge. @ojaswa1942 has already approved this one.

**Review by khemrajrathore — APPROVED** [review]
LGTM

**Pull request merged**